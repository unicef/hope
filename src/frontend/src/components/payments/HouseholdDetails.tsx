import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { Overview } from '@components/payments/Overview';
import { Title } from '@core/Title';
import { Grid, Typography } from '@mui/material';
import { LabelizedField } from '@core/LabelizedField';
import { getPhoneNoLabel } from '@utils/utils';

export const HouseholdDetails = ({ household }): React.ReactElement => {
  const { t } = useTranslation();
  return (
    <Overview>
      <Title>
        <Typography variant="h6">{t('Household')}</Typography>
      </Title>
      <Grid container spacing={3}>
        <Grid item xs={3}>
          <LabelizedField
            label={t('HOUSEHOLD ID')}
            value={household.unicefId}
          />
        </Grid>
        <Grid item xs={3}>
          <LabelizedField
            label={t('HEAD OF HOUSEHOLD')}
            value={household.headOfHousehold.fullName}
          />
        </Grid>
        <Grid item xs={3}>
          <LabelizedField
            label={t('TOTAL PERSON COVERED')}
            value={household.size}
          />
        </Grid>
        <Grid item xs={3}>
          <LabelizedField
            label={t('PHONE NUMBER')}
            value={getPhoneNoLabel(
              household.headOfHousehold.phoneNo,
              household.headOfHousehold.phoneNoValid,
            )}
          />
        </Grid>
        <Grid item xs={3}>
          <LabelizedField
            label={t('ALT. PHONE NUMBER')}
            value={getPhoneNoLabel(
              household.headOfHousehold.phoneNoAlternative,
              household.headOfHousehold.phoneNoAlternativeValid,
            )}
          />
        </Grid>
      </Grid>
    </Overview>
  );
};
