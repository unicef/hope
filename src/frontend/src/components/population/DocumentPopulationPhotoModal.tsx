import * as React from 'react';
import { IndividualNode, useIndividualPhotosQuery } from '@generated/graphql';
import { PhotoModal } from '@core/PhotoModal/PhotoModal';

interface DocumentPopulationPhotoModalProps {
  individual: IndividualNode;
  documentNumber: string;
  documentId: string;
}

export function DocumentPopulationPhotoModal({
  individual,
  documentNumber,
  documentId,
}: DocumentPopulationPhotoModalProps): React.ReactElement {
  const { data } = useIndividualPhotosQuery({
    variables: { id: individual?.id },
    fetchPolicy: 'network-only',
  });
  const documentWithPhoto = data?.individual?.documents?.edges?.find(
    (el) => el.node.id === documentId,
  );

  return (
    <PhotoModal
      src={documentWithPhoto?.node?.photo}
      linkText={documentNumber}
      variant="link"
      title="Document's Photo"
    />
  );
}
