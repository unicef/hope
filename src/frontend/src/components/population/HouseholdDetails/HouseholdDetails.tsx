import { Box, Grid, Paper, Typography } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import {
  GrievancesChoiceDataQuery,
  HouseholdChoiceDataQuery,
  HouseholdNode,
} from '@generated/graphql';
import { useProgramContext } from '../../../programContext';
import { choicesToDict, formatCurrencyWithSymbol } from '@utils/utils';
import { ContentLink } from '@core/ContentLink';
import { LabelizedField } from '@core/LabelizedField';
import { Title } from '@core/Title';
import {
  BigValue,
  BigValueContainer,
} from '../../rdi/details/RegistrationDetails/RegistrationDetails';
import { LinkedGrievancesModal } from '../LinkedGrievancesModal/LinkedGrievancesModal';

const Container = styled.div`
  display: flex;
  flex: 1;
  width: 100%;
  background-color: #fff;
  padding: ${({ theme }) => theme.spacing(8)}
    ${({ theme }) => theme.spacing(11)};
  flex-direction: column;
  align-items: center;
  border-color: #b1b1b5;
  border-bottom-width: 1px;
  border-bottom-style: solid;

  && > div {
    margin: 5px;
  }
`;

const Overview = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`;
const OverviewPaper = styled(Paper)`
  margin: 20px 20px 0 20px;
  padding: 20px ${({ theme }) => theme.spacing(11)};
`;

interface HouseholdDetailsProps {
  household: HouseholdNode;
  choicesData: HouseholdChoiceDataQuery;
  baseUrl: string;
  businessArea: string;
  grievancesChoices: GrievancesChoiceDataQuery;
}
export function HouseholdDetails({
  household,
  choicesData,
  baseUrl,
  businessArea,
  grievancesChoices,
}: HouseholdDetailsProps): React.ReactElement {
  const { t } = useTranslation();
  const residenceChoicesDict = choicesToDict(
    choicesData.residenceStatusChoices,
  );
  const { selectedProgram } = useProgramContext();
  return (
    <>
      <Container>
        <Title>
          <Typography variant="h6">{t('Details')}</Typography>
        </Title>
        <Overview>
          <Grid container spacing={3}>
            <Grid item xs={3}>
              <LabelizedField label={t('Household Size')}>
                {household?.size}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Residence Status')}>
                {residenceChoicesDict[household?.residenceStatus]}
              </LabelizedField>
            </Grid>
            <Grid item xs={6}>
              <LabelizedField label={t('Head of Household')}>
                <ContentLink
                  href={`/${baseUrl}/population/individuals/${household?.headOfHousehold?.id}`}
                >
                  {household?.headOfHousehold?.fullName}
                </ContentLink>
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('FEMALE CHILD HEADED HOUSEHOLD')}>
                {household?.fchildHoh ? t('Yes') : t('No')}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('CHILD HEADED HOUSEHOLD')}>
                {household?.childHoh ? t('Yes') : t('No')}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Country')}>
                {household?.country}
              </LabelizedField>
            </Grid>

            <Grid item xs={3}>
              <LabelizedField label={t('Country of Origin')}>
                {household?.countryOrigin}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Address')}>
                {household?.address}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Village')}>
                {household?.village}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Zip code')}>
                {household?.zipCode}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Administrative Level 1')}>
                {household?.admin1?.name}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Administrative Level 2')}>
                {household?.admin2?.name}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Administrative Level 3')}>
                {household?.admin3?.name}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Administrative Level 4')}>
                {household?.admin4?.name}
              </LabelizedField>
            </Grid>
            <Grid item xs={6}>
              <LabelizedField label={t('Geolocation')}>
                {household?.geopoint
                  ? `${household?.geopoint?.coordinates[0]}, ${household?.geopoint?.coordinates[1]}`
                  : '-'}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('UNHCR CASE ID')}>
                {household?.unhcrId}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('LENGTH OF TIME SINCE ARRIVAL')}>
                {household?.flexFields?.months_displaced_h_f}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('NUMBER OF TIMES DISPLACED')}>
                {household?.flexFields?.number_times_displaced_h_f}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('IS THIS A RETURNEE HOUSEHOLD?')}>
                {household?.returnee ? t('Yes') : t('No')}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              {household?.unicefId && (
                <LinkedGrievancesModal
                  household={household}
                  baseUrl={baseUrl}
                  businessArea={businessArea}
                  grievancesChoices={grievancesChoices}
                />
              )}
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Data Collecting Type')}>
                {selectedProgram?.dataCollectingType?.label}
              </LabelizedField>
            </Grid>
          </Grid>
        </Overview>
      </Container>
      <OverviewPaper>
        <Title>
          <Typography variant="h6">{t('Benefits')}</Typography>
        </Title>
        <Grid container>
          <Grid item xs={3}>
            <LabelizedField label={t('Cash received')}>
              {household?.deliveredQuantities?.length ? (
                <Box mb={2}>
                  <Grid container>
                    <Grid item xs={6}>
                      <Box display="flex" flexDirection="column">
                        {household?.deliveredQuantities?.map((item) => (
                          <Box
                            key={`${item.currency}-${item.totalDeliveredQuantity}`}
                          >
                            {item.currency === 'USD'
                              ? formatCurrencyWithSymbol(
                                  item.totalDeliveredQuantity,
                                  item.currency,
                                )
                              : `(${formatCurrencyWithSymbol(
                                  item.totalDeliveredQuantity,
                                  item.currency,
                                )})`}
                          </Box>
                        ))}
                      </Box>
                    </Grid>
                  </Grid>
                </Box>
              ) : (
                <>-</>
              )}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <BigValueContainer>
              <LabelizedField label={t('Total Cash Received')}>
                <BigValue>
                  {formatCurrencyWithSymbol(
                    household?.totalCashReceivedUsd,
                    'USD',
                  )}
                </BigValue>
              </LabelizedField>
            </BigValueContainer>
          </Grid>
        </Grid>
      </OverviewPaper>
    </>
  );
}
