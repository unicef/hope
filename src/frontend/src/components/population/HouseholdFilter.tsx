import { Grid, MenuItem } from '@mui/material';
import AssignmentIndRoundedIcon from '@mui/icons-material/AssignmentIndRounded';
import FlashOnIcon from '@mui/icons-material/FlashOn';
import GroupIcon from '@mui/icons-material/Group';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation, useNavigate } from 'react-router-dom';
import { HouseholdChoiceDataQuery, ProgramNode } from '@generated/graphql';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { AdminAreaAutocomplete } from '@shared/autocompletes/AdminAreaAutocomplete';
import { householdTableOrderOptions } from '@utils/constants';
import { createHandleApplyFilterChange } from '@utils/utils';
import { FiltersSection } from '@core/FiltersSection';
import { NumberTextField } from '@core/NumberTextField';
import { SearchTextField } from '@core/SearchTextField';
import { SelectFilter } from '@core/SelectFilter';
import { DocumentSearchField } from '@core/DocumentSearchField';

interface HouseholdFiltersProps {
  filter;
  programs?: ProgramNode[];
  choicesData: HouseholdChoiceDataQuery;
  setFilter: (filter) => void;
  initialFilter;
  appliedFilter;
  setAppliedFilter: (filter) => void;
  isOnPaper?: boolean;
}

export function HouseholdFilters({
  filter,
  programs,
  choicesData,
  setFilter,
  initialFilter,
  appliedFilter,
  setAppliedFilter,
  isOnPaper = true,
}: HouseholdFiltersProps): React.ReactElement {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const location = useLocation();
  const { isAllPrograms } = useBaseUrl();
  const { handleFilterChange, applyFilterChanges, clearFilter } =
    createHandleApplyFilterChange(
      initialFilter,
      navigate,
      location,
      filter,
      setFilter,
      appliedFilter,
      setAppliedFilter,
    );
  const handleApplyFilter = (): void => {
    applyFilterChanges();
  };

  const handleClearFilter = (): void => {
    clearFilter();
  };

  return (
    <FiltersSection
      clearHandler={handleClearFilter}
      applyHandler={handleApplyFilter}
      isOnPaper={isOnPaper}
    >
      <Grid container alignItems="flex-end" spacing={3}>
        <Grid item xs={3}>
          <SearchTextField
            label={t('Search')}
            value={filter.search}
            onChange={(e) => handleFilterChange('search', e.target.value)}
            data-cy="hh-filters-search"
          />
        </Grid>
        <DocumentSearchField
          onChange={handleFilterChange}
          type={filter.documentType}
          number={filter.documentNumber}
          choices={choicesData?.documentTypeChoices}
        />
        {isAllPrograms && (
          <Grid item xs={3}>
            <SelectFilter
              onChange={(e) => handleFilterChange('program', e.target.value)}
              label={t('Programme')}
              value={filter.program}
              fullWidth
              icon={<FlashOnIcon />}
              data-cy="hh-filters-program"
            >
              {programs.map((program) => (
                <MenuItem key={program.id} value={program.id}>
                  {program.name}
                </MenuItem>
              ))}
            </SelectFilter>
          </Grid>
        )}
        <Grid item xs={3}>
          <SelectFilter
            onChange={(e) =>
              handleFilterChange('residenceStatus', e.target.value)
            }
            label={t('Residence Status')}
            fullWidth
            value={filter.residenceStatus}
            icon={<AssignmentIndRoundedIcon />}
            data-cy="hh-filters-residence-status"
          >
            {choicesData.residenceStatusChoices?.map((status) => (
              <MenuItem key={status.value} value={status.value}>
                {status.name}
              </MenuItem>
            ))}
          </SelectFilter>
        </Grid>
        <Grid item xs={3}>
          <NumberTextField
            topLabel={t('Household Size')}
            value={filter.householdSizeMin}
            placeholder={t('From')}
            icon={<GroupIcon />}
            fullWidth
            onChange={(e) =>
              handleFilterChange('householdSizeMin', e.target.value)
            }
            data-cy="hh-filters-household-size-from"
          />
        </Grid>
        <Grid item xs={3}>
          <NumberTextField
            value={filter.householdSizeMax}
            placeholder={t('To')}
            icon={<GroupIcon />}
            fullWidth
            onChange={(e) =>
              handleFilterChange('householdSizeMax', e.target.value)
            }
            data-cy="hh-filters-household-size-to"
          />
        </Grid>
        <Grid item xs={3}>
          <SelectFilter
            onChange={(e) => handleFilterChange('orderBy', e.target.value)}
            label={t('Sort by')}
            value={filter.orderBy}
            data-cy="hh-filters-order-by"
            disableClearable
          >
            {householdTableOrderOptions.map((order) => (
              <MenuItem key={order.value} value={order.value}>
                {order.name}
              </MenuItem>
            ))}
          </SelectFilter>
        </Grid>
        <Grid item xs={3}>
          <SelectFilter
            onChange={(e) => handleFilterChange('withdrawn', e.target.value)}
            label={t('Status')}
            value={filter.withdrawn}
            data-cy="hh-filters-status"
          >
            <MenuItem key="all" value="null">
              All
            </MenuItem>
            <MenuItem key="active" value="false">
              Active
            </MenuItem>
            <MenuItem key="inactive" value="true">
              Withdrawn
            </MenuItem>
          </SelectFilter>
        </Grid>
        {isAllPrograms && (
          <Grid item xs={3}>
            <SelectFilter
              onChange={(e) =>
                handleFilterChange('programState', e.target.value)
              }
              label={t('Programme State')}
              value={filter.programState}
              fullWidth
              disableClearable
              data-cy="filters-program-state"
            >
              <MenuItem value="active">{t('Active Programmes')}</MenuItem>
              <MenuItem value="all">{t('All Programmes')}</MenuItem>
            </SelectFilter>
          </Grid>
        )}
        <Grid item xs={3}>
          <AdminAreaAutocomplete
            name="admin1"
            level={1}
            value={filter.admin1}
            filter={filter}
            setFilter={setFilter}
            initialFilter={initialFilter}
            appliedFilter={appliedFilter}
            setAppliedFilter={setAppliedFilter}
            disabled={filter.admin2}
            dataCy="hh-filters-admin1"
          />
        </Grid>
        <Grid item xs={3}>
          <AdminAreaAutocomplete
            name="admin2"
            level={2}
            value={filter.admin2}
            filter={filter}
            setFilter={setFilter}
            initialFilter={initialFilter}
            appliedFilter={appliedFilter}
            setAppliedFilter={setAppliedFilter}
            disabled={filter.admin1}
            dataCy="hh-filters-admin2"
          />
        </Grid>
      </Grid>
    </FiltersSection>
  );
}
