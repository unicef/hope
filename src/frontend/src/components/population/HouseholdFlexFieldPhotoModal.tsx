import { Box } from '@mui/material';
import * as React from 'react';
import { useParams } from 'react-router-dom';
import { useHouseholdFlexFieldsQuery } from '@generated/graphql';
import { PhotoModal } from '@core/PhotoModal/PhotoModal';

export function HouseholdFlexFieldPhotoModal({ field }): React.ReactElement {
  const { id } = useParams();
  const { data } = useHouseholdFlexFieldsQuery({
    variables: { id },
    fetchPolicy: 'network-only',
  });

  if (!data) {
    return null;
  }

  const { flexFields } = data.household;
  const picUrl = flexFields[field.name];

  return picUrl ? (
    <PhotoModal src={picUrl} />
  ) : (
    <Box display="flex" alignItems="center">
      -
    </Box>
  );
}
