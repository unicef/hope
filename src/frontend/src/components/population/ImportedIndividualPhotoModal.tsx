import * as React from 'react';
import {
  ImportedIndividualNode,
  useImportedIndividualPhotosQuery,
} from '@generated/graphql';
import { PhotoModal } from '@core/PhotoModal/PhotoModal';

interface ImportedIndividualPhotoModalProps {
  individual: ImportedIndividualNode;
}

export function ImportedIndividualPhotoModal({
  individual,
}: ImportedIndividualPhotoModalProps): React.ReactElement {
  const { data } = useImportedIndividualPhotosQuery({
    variables: { id: individual?.id },
    fetchPolicy: 'network-only',
  });

  return (
    <PhotoModal
      src={data?.importedIndividual?.photo}
      variant="button"
      title="Individuals's Photo"
    />
  );
}
