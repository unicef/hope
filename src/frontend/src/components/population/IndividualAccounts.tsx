import { DividerLine } from '@components/core/DividerLine';
import { HouseholdChoiceDataQuery, IndividualNode } from '@generated/graphql';
import { Grid2 as Grid, Paper, Typography } from '@mui/material';
import { Title } from '@core/Title';
import { t } from 'i18next';
import React, { FC } from 'react';
import styled from 'styled-components';
import { LabelizedField } from '@components/core/LabelizedField';
import { Title } from '@core/Title';
import { usePermissions } from '@hooks/usePermissions';
import { Grid2 as Grid, Paper, Theme, Typography } from '@mui/material';
import { IndividualDetail } from '@restgenerated/models/IndividualDetail';
import { renderSomethingOrDash } from '@utils/utils';
import { t } from 'i18next';
import { FC } from 'react';
import { hasPermissions, PERMISSIONS } from 'src/config/permissions';
import { useProgramContext } from 'src/programContext';
import styled from 'styled-components';
import { useArrayToDict } from '@hooks/useArrayToDict';

interface IndividualAccountsProps {
  individual: IndividualDetail;
  choicesData: HouseholdChoiceDataQuery;
}

const Overview = styled(Paper)<{ theme?: Theme }>`
  padding: ${({ theme }) => theme.spacing(8)}
    ${({ theme }) => theme.spacing(11)};
  margin-top: ${({ theme }) => theme.spacing(6)};
  margin-bottom: ${({ theme }) => theme.spacing(4)};
`;

export const IndividualAccounts: FC<IndividualAccountsProps> = ({
  individual,
  choicesData,
}) => {
  const permissions = usePermissions();
  const canViewDeliveryMechanisms = hasPermissions(
    PERMISSIONS.POPULATION_VIEW_INDIVIDUAL_DELIVERY_MECHANISMS_SECTION,
    permissions,
  );
  const { selectedProgram } = useProgramContext();
  const beneficiaryGroup = selectedProgram?.beneficiaryGroup;
  const accountFinancialInstitutionsDict = useArrayToDict(
    choicesData.accountFinancialInstitutionChoices,
    'value',
    'name',
  );

  if (!individual?.accounts?.length || !canViewDeliveryMechanisms) {
    return null;
  }
  return (
    <Overview>
      <Title>
        <Typography variant="h6">
          {t(`${beneficiaryGroup?.memberLabel} Accounts`)}
        </Typography>
      </Title>
      <Grid container spacing={6}>
        {individual.accounts.map((mechanism, index) => {
          const tabData = JSON.parse(mechanism.data);
          return (
            <Grid size={{ xs: 12 }} key={index}>
              <Typography variant="h6">{mechanism.accountType}</Typography>
              <Grid container spacing={3}>
                {Object.entries(tabData).map(([key, value], idx) => {
                  if (key === 'financial_institution') {
                    value = accountFinancialInstitutionsDict[value as string];
                  }
                  return (
                  <Grid key={idx} size={{ xs: 3 }}>
                    <LabelizedField label={key.replace(/_/g, ' ')}>
                      {renderSomethingOrDash(value)}
                    </LabelizedField>
                  </Grid>
                );
                })}
              </Grid>
              {index < individual.accounts.length - 1 && <DividerLine />}
            </Grid>
          );
        })}
      </Grid>
    </Overview>
  );
};
