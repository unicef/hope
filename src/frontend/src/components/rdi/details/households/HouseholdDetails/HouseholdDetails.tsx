import { Grid, Typography } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { ContainerColumnWithBorder } from '@core/ContainerColumnWithBorder';
import { ContentLink } from '@core/ContentLink';
import { LabelizedField } from '@core/LabelizedField';
import { choicesToDict } from '@utils/utils';
import {
  HouseholdChoiceDataQuery,
  ImportedHouseholdDetailedFragment,
} from '@generated/graphql';
import { Title } from '@core/Title';
import { useProgramContext } from '../../../../../programContext';
import { useBaseUrl } from '@hooks/useBaseUrl';

const Overview = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`;

interface HouseholdDetailsProps {
  household: ImportedHouseholdDetailedFragment;
  choicesData: HouseholdChoiceDataQuery;
}
export function HouseholdDetails({
  household,
  choicesData,
}: HouseholdDetailsProps): React.ReactElement {
  const { baseUrl } = useBaseUrl();
  const { t } = useTranslation();
  const { selectedProgram } = useProgramContext();
  const residenceChoicesDict = choicesToDict(
    choicesData.residenceStatusChoices,
  );
  return (
    <ContainerColumnWithBorder>
      <Title>
        <Typography variant="h6">{t('Details')}</Typography>
      </Title>
      <Overview>
        <Grid container spacing={6}>
          <Grid item xs={3}>
            <LabelizedField label={t('Household Size')}>
              {household.size}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('Country')}>
              {household.country}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('Residence Status')}>
              {residenceChoicesDict[household.residenceStatus]}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('Country of Origin')}>
              {household.countryOrigin}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('Head of Household')}>
              <ContentLink
                href={`/${baseUrl}/registration-data-import/individual/${household.headOfHousehold.id}`}
              >
                {household.headOfHousehold.fullName}
              </ContentLink>
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('FEMALE CHILD HEADED HOUSEHOLD')}>
              {household.fchildHoh ? 'Yes' : 'No'}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('CHILD HEADED HOUSEHOLD')}>
              {household.childHoh ? 'Yes' : 'No'}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('ADMINISTRATIVE LEVEL 1')}>
              {household.admin1?.name}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('ADMINISTRATIVE LEVEL 2')}>
              {household.admin2?.name}
            </LabelizedField>
          </Grid>
          <Grid item xs={3}>
            <LabelizedField label={t('Data Collecting Type')}>
              {selectedProgram?.dataCollectingType?.label}
            </LabelizedField>
          </Grid>
        </Grid>
      </Overview>
    </ContainerColumnWithBorder>
  );
}
