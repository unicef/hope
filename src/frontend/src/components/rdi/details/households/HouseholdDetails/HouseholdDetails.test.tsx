import * as React from 'react';
import { fakeHouseholdChoices } from '../../../../../../fixtures/population/fakeHouseholdChoices';
import { fakeImportedHouseholdDetailedFragment } from '../../../../../../fixtures/registration/fakeImportedHouseholdDetailedFragment';
import { render } from '../../../../../testUtils/testUtils';
import { HouseholdDetails } from './HouseholdDetails';

describe('components/rdi/details/households/HouseholdDetails', () => {
  it('should render', () => {
    const { container } = render(
      <HouseholdDetails
        household={fakeImportedHouseholdDetailedFragment}
        choicesData={fakeHouseholdChoices}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
