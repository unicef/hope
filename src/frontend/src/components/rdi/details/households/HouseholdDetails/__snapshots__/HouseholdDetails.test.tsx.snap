// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/rdi/details/households/HouseholdDetails should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-beySPh fvkwGO css-1ps6pg7-MuiPaper-root"
  >
    <div
      class="sc-eDLKkx bKQhaX"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
      >
        Details
      </h6>
    </div>
    <div
      class="sc-jTQCzO iuuSmf"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              Household Size
            </span>
            <div
              data-cy="label-Household Size"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                4
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              Country
            </span>
            <div
              data-cy="label-Country"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                Isle of Man
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              Residence Status
            </span>
            <div
              data-cy="label-Residence Status"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                Displaced  |  Refugee / Asylum Seeker
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              Country of Origin
            </span>
            <div
              data-cy="label-Country of Origin"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                San Marino
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              Head of Household
            </span>
            <div
              data-cy="label-Head of Household"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                <a
                  class="sc-guDLey ibjlhn"
                  href="/afghanistan/programs/UHJvZ3JhbU5vZGU6YzRkNTY1N2QtMWEyOS00NmUxLTgxOTAtZGY3Zjg1YTBkMmVm/registration-data-import/individual/SW1wb3J0ZWRJbmRpdmlkdWFsTm9kZTpmM2U0ODBiZC05MDYyLTRkMjktODljYy1kMWM3MGRhZDRmMjk="
                >
                  Agata Kowalska
                </a>
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              FEMALE CHILD HEADED HOUSEHOLD
            </span>
            <div
              data-cy="label-FEMALE CHILD HEADED HOUSEHOLD"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                No
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              CHILD HEADED HOUSEHOLD
            </span>
            <div
              data-cy="label-CHILD HEADED HOUSEHOLD"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                No
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              ADMINISTRATIVE LEVEL 1
            </span>
            <div
              data-cy="label-ADMINISTRATIVE LEVEL 1"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                Nangarhar
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              ADMINISTRATIVE LEVEL 2
            </span>
            <div
              data-cy="label-ADMINISTRATIVE LEVEL 2"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                Achin
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-dmyCSP edirvw"
              color="textSecondary"
            >
              Data Collecting Type
            </span>
            <div
              data-cy="label-Data Collecting Type"
            >
              <span
                class="sc-hLQSwg hXyiIZ"
                color="textSecondary"
              >
                data collecting type
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
